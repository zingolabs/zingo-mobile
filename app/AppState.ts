// new class with all the information about the sync process
export class SyncStatusReport {
  // sync ID
  syncID: number;

  // total of batches to process (usually 1000 blocks per batch)
  totalBatches: number;

  // batch that is processing now
  currentBatch: number;

  // total blocks per batch
  blocksPerBatch: number;

  // last block of the wallet
  lastBlockWallet: number;

  // block that is processing
  currentBlock: number;

  inProgress: boolean;

  lastError: string;

  secondsPerBatch: number;

  percent: number;

  message: string;

  process_end_block: number;

  lastBlockServer: number;

  constructor() {
    this.syncID = 0;
    this.totalBatches = 0;
    this.currentBatch = 0;
    this.blocksPerBatch = 1000; // this is set in zingolib
    this.lastBlockWallet = 0;
    this.currentBlock = 0;
    this.inProgress = false;
    this.lastError = '';
    this.secondsPerBatch = 0;
    this.percent = 0;
    this.message = '';
    this.process_end_block = 0;
    this.lastBlockServer = 0;
  }
}

export class TotalBalance {
  // Total t address, confirmed and spendable
  transparentBal: number;

  // Total private, confirmed + unconfirmed
  privateBal: number;

  // Total orchard, confirmed + unconfirmed
  orchardBal: number;

  // Total private, confirmed funds that are spendable
  spendablePrivate: number;

  // Total orchard, confirmed funds that are spendable
  spendableOrchard: number;

  // Total unconfirmed + spendable
  total: number;

  constructor() {
    this.transparentBal = 0;
    this.privateBal = 0;
    this.spendablePrivate = 0;
    this.orchardBal = 0;
    this.spendableOrchard = 0;
    this.total = 0;
  }
}

export class Address {
  uaAddress: string;
  address: string;
  addressKind: string;
  containsPending: boolean;

  constructor(uaAddress: string, address: string, addressKind: string) {
    this.uaAddress = uaAddress;
    this.address = address;
    this.addressKind = addressKind;
    this.containsPending = false;
  }
}

export class AddressBookEntry {
  label: string;

  address: string;

  constructor(label: string, address: string) {
    this.label = label;
    this.address = address;
  }
}

export class SettingsFileEntry {
  server?: string;

  constructor(server: string) {
    this.server = server;
  }
}

export interface TxDetailType {
  address: string;
  amount: number;
  memo: string | null;
}

// List of transactions.
export interface Transaction {
  type: string;
  address: string;
  amount: number;
  position: string;
  confirmations: number;
  txid: string;
  time: number;
  zec_price: number | null;
  detailedTxns: TxDetailType[];
}

// This is the type that the RPC interface expects for sending
export interface SendJsonToType {
  address: string;
  amount: number;
  memo?: string;
}

export class ToAddr {
  id: number;
  to: string;
  amount: string;
  amountUSD: string;
  memo: string;

  constructor(id: number) {
    this.id = id;

    this.to = '';
    this.amount = '';
    this.amountUSD = '';
    this.memo = '';
  }
}

export class SendPageState {
  toaddrs: ToAddr[];

  constructor() {
    this.toaddrs = [];
  }
}

export class ReceivePageState {
  // A newly created address to show by default
  newAddress: string;

  // The key used for the receive page component.
  // Increment to force re-render
  rerenderKey: number;

  constructor() {
    this.newAddress = '';
    this.rerenderKey = 0;
  }
}

export interface InfoType {
  chain_name: string;
  serverUri: string;
  latestBlock: number;
  connections: number;
  version: string;
  verificationProgress: number;
  currencyName: string;
  solps: number;
  zecPrice: number | null;
  defaultFee: number;
}

export class SendProgress {
  sendInProgress: boolean;
  progress: number;
  total: number;
  etaSeconds: number | string;

  constructor() {
    this.sendInProgress = false;
    this.progress = 0;
    this.total = 0;
    this.etaSeconds = 0;
  }
}

export interface ServerSelectState {
  modalIsOpen: boolean;
}

export class PasswordState {
  showPassword: boolean;

  confirmNeeded: boolean;

  passwordCallback: ((password: string) => void) | null;

  closeCallback: (() => void) | null;

  helpText: string | null;

  constructor() {
    this.showPassword = false;
    this.confirmNeeded = false;
    this.passwordCallback = null;
    this.closeCallback = null;
    this.helpText = null;
  }
}

export class ErrorModalData {
  title: string;
  body: string;

  modalIsOpen: boolean;

  constructor() {
    this.modalIsOpen = false;
    this.title = '';
    this.body = '';
  }
}

export interface ConnectedCompanionApp {
  name: string;
  lastSeen: number;
}

export interface WalletSeed {
  seed: string;
  birthday: number;
}

export interface SyncStatus {
  inProgress: boolean;
  progress: number;
  blocks: string;
}

export class WalletSettings {
  download_memos: string;
  transaction_filter_threshold: string;
  server?: string;

  constructor() {
    this.download_memos = 'wallet';
    this.transaction_filter_threshold = '500';
  }
}

export default interface AppState {
  // Info about the current sync process
  syncStatusReport: SyncStatusReport;

  // The total confirmed and unconfirmed balance in this wallet
  totalBalance: TotalBalance;

  // A map type that contains address -> privatekey mapping, for display on the receive page
  // This mapping is ephemeral, and will disappear when the user navigates away.
  addressPrivateKeys: Map<string, string>;

  // List of all addresses in the wallet, including change addresses and addresses
  // that don't have any balance or are unused
  addresses: Address[];

  // List of Address / Label pairs
  addressBook: AddressBookEntry[];

  // List of all T and Z and O transactions
  transactions: Transaction[] | null;

  // The state of the send page, as the user constructs a transaction
  sendPageState: SendPageState;

  // Any state for the receive page
  receivePageState: ReceivePageState;

  // getinfo and getblockchaininfo result
  info: InfoType | null;

  // Is the app rescanning?
  rescanning: boolean;

  // Callbacks for the password dialog box
  //passwordState: PasswordState;

  wallet_settings: WalletSettings;

  syncingStatus: SyncStatus | null;

  // Data for any error or info modal
  errorModalData: ErrorModalData;

  // Build progress from Tx
  txBuildProgress: SendProgress;

  walletSeed: WalletSeed | null;

  isMenuDrawerOpen: boolean;

  selectedMenuDrawerItem: string;

  aboutModalVisible: boolean;

  computingModalVisible: boolean;

  settingsModalVisible: boolean;

  infoModalVisible: boolean;

  rescanModalVisible: boolean;

  seedViewModalVisible: boolean;
  seedChangeModalVisible: boolean;
  seedBackupModalVisible: boolean;
  seedServerModalVisible: boolean;

  syncReportModalVisible: boolean;

  newServer: string | null;

  uaAddress: string | null;

  // eslint-disable-next-line semi
}
