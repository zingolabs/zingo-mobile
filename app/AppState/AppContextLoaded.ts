import { StackScreenProps } from '@react-navigation/stack';

import TotalBalanceClass from './classes/TotalBalanceClass';
import AddressClass from './classes/AddressClass';
import SendPageStateClass from './classes/SendPageStateClass';
import SendProgressClass from './classes/SendProgressClass';
import ReceivePageStateClass from './classes/ReceivePageStateClass';
import WalletSettingsClass from './classes/WalletSettingsClass';
import AddressBookFileClass from './classes/AddressBookFileClass';
import SyncingStatusClass from './classes/SyncingStatusClass';

import InfoType from './types/InfoType';
import WalletType from './types/WalletType';
import ZecPriceType from './types/ZecPriceType';
import BackgroundType from './types/BackgroundType';
import { TranslateType } from './types/TranslateType';
import NetInfoType from './types/NetInfoType';
import BackgroundErrorType from './types/BackgroundErrorType';
import ServerType from './types/ServerType';
import SnackbarType from './types/SnackbarType';
import SecurityType from './types/SecurityType';

import { LanguageEnum } from './enums/LanguageEnum';
import { CurrencyEnum } from './enums/CurrencyEnum';
import { ModeEnum } from './enums/ModeEnum';
import { SelectServerEnum } from './enums/SelectServerEnum';
import ValueTransferType from './types/ValueTransferType';

export default interface AppContextLoaded {
  navigation: StackScreenProps<any>['navigation'];
  netInfo: NetInfoType;

  // The total confirmed and pending balance in this wallet
  totalBalance: TotalBalanceClass | null;

  // List of all addresses in the wallet, including change addresses and addresses
  // that don't have any balance or are unused
  addresses: AddressClass[] | null;

  // List of all T and Z and O value transfers
  valueTransfers: ValueTransferType[] | null;

  // The state of the send page
  sendPageState: SendPageStateClass;

  // Any state for the receive page
  receivePageState: ReceivePageStateClass;

  // getinfo and getblockchaininfo result
  info: InfoType;

  // internal wallet settings from blockchain
  walletSettings: WalletSettingsClass;

  // syncing Info about the status of the process
  syncingStatus: SyncingStatusClass;

  // Build progress from Tx
  sendProgress: SendProgressClass;

  // wallet recovery info
  wallet: WalletType;

  // active UA in the wallet
  uaAddress: string;

  // zec price in USD from internet
  zecPrice: ZecPriceType;

  // info about background syncing
  background: BackgroundType;

  // helper to get text tranalated to the active language
  translate: (key: string) => TranslateType;

  // Error from the App when is in background
  backgroundError: BackgroundErrorType;
  setBackgroundError: (title: string, error: string) => void;

  // this wallet is watch-only (Readonly)
  readOnly: boolean;

  // settings
  server: ServerType;
  currency: CurrencyEnum;
  language: LanguageEnum;
  sendAll: boolean;
  donation: boolean;
  privacy: boolean;
  mode: ModeEnum;
  security: SecurityType;
  selectServer: SelectServerEnum;
  rescanMenu: boolean;
  recoveryWalletInfoOnDevice: boolean;

  // snackbar queue
  snackbars: SnackbarType[];
  addLastSnackbar: (snackbar: SnackbarType) => void;

  // if the App is stalled - restart is fired
  restartApp: (s: any) => void;

  // some ValueTransfer is pending?
  somePending: boolean;

  // List of our contacts - Address book
  addressBook: AddressBookFileClass[];

  // helpers to open the address book modal from different places in the App
  launchAddressBook: (add: string, close: () => void, open: () => void) => void;
  addressBookCurrentAddress: string;
  addressBookOpenPriorModal: () => void;

  // is calculated in the header & needed in the send screen
  shieldingAmount: number;

  // eslint-disable-next-line semi
}
