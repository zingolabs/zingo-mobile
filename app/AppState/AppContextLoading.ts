import { StackScreenProps } from '@react-navigation/stack';

import InfoType from './types/InfoType';
import WalletType from './types/WalletType';
import ZecPriceType from './types/ZecPriceType';
import BackgroundType from './types/BackgroundType';
import { TranslateType } from './types/TranslateType';
import NetInfoType from './types/NetInfoType';
import BackgroundErrorType from './types/BackgroundErrorType';
import ServerType from './types/ServerType';
import SnackbarType from './types/SnackbarType';
import SecurityType from './types/SecurityType';

import { CurrencyEnum } from './enums/CurrencyEnum';
import { LanguageEnum } from './enums/LanguageEnum';
import { ModeEnum } from './enums/ModeEnum';
import { SelectServerEnum } from './enums/SelectServerEnum';

export default interface AppContextLoading {
  // context
  navigation: StackScreenProps<any>['navigation'];
  netInfo: NetInfoType;
  wallet: WalletType;
  info: InfoType;
  zecPrice: ZecPriceType;
  background: BackgroundType;
  translate: (key: string) => TranslateType;
  backgroundError: BackgroundErrorType;
  setBackgroundError: (title: string, error: string) => void;
  readOnly: boolean;
  snackbars: SnackbarType[];
  addLastSnackbar: (snackbar: SnackbarType) => void;

  // context settings
  server: ServerType;
  currency: CurrencyEnum;
  language: LanguageEnum;
  sendAll: boolean;
  donation: boolean;
  privacy: boolean;
  mode: ModeEnum;
  security: SecurityType;
  selectServer: SelectServerEnum;
  rescanMenu: boolean;
  recoveryWalletInfoOnDevice: boolean;

  // eslint-disable-next-line semi
}
