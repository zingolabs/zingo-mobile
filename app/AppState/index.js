import AddressClass from './classes/AddressClass';
import AddressBookClass from './classes/AddressBookClass';
import ErrorModalDataClass from './classes/ErrorModalDataClass';
import ReceivePageStateClass from './classes/ReceivePageStateClass';
import SendPageStateClass from './classes/SendPageStateClass';
import SendProgressClass from './classes/SendProgressClass';
import SettingsFileClass from './classes/SettingsFileClass';
import SyncingStatusClass from './classes/SyncingStatusClass';
import ToAddrClass from './classes/ToAddrClass';
import TotalBalanceClass from './classes/TotalBalanceClass';
import WalletSettingsClass from './classes/WalletSettingsClass';

import InfoType from './types/InfoType';
import SendJsonToTypeType from './types/SendJsonToTypeType';
import TransactionType from './types/TransactionType';
import TxDetailType from './types/TxDetailType';
import WalletType from './types/WalletType';
import { TranslateType } from './types/TranslateType';
import NetInfoType from './types/NetInfoType';
import BackgroundType from './types/BackgroundType';
import BackgroundErrorType from './types/BackgroundErrorType';
import ServerType from './types/ServerType';
import SnackbarType from './types/SnackbarType';
import ZecPriceType from './types/ZecPriceType';
import FetchErrorType from './types/FetchErrorType';

import AppStateLoaded from './AppStateLoaded';
import AppStateLoading from './AppStateLoading';

export {
  AppStateLoaded,
  AppStateLoading,
  AddressClass,
  AddressBookClass,
  ErrorModalDataClass,
  ReceivePageStateClass,
  SendPageStateClass,
  SendProgressClass,
  SettingsFileClass,
  SyncingStatusClass,
  ToAddrClass,
  TotalBalanceClass,
  WalletSettingsClass,
  TransactionType,
  TxDetailType,
  WalletType,
  SendJsonToTypeType,
  InfoType,
  ZecPriceType,
  BackgroundType,
  BackgroundErrorType,
  TranslateType,
  NetInfoType,
  ServerType,
  SnackbarType,
  FetchErrorType,
};
