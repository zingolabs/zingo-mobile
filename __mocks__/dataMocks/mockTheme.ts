import { ThemeType } from '../../app/types';

export const mockTheme: ThemeType = {
  dark: true,
  colors: {
    background: '#011401', //'#010101',
    card: '#011401', //'#401717',
    border: '#ffffff',
    primary: '#18bd18', //'#df4100',
    primaryDisabled: '#5a8c5a', //'rgba(90, 140, 90, 1)',
    secondaryDisabled: '#233623',
    text: '#c3c3c3',
    zingo: '#888888',
    placeholder: '#888888',
    money: '#ffffff',
    syncing: '#ebff5a',
    notification: '',
    sideMenuBackground: '#040C17',
  },
};
