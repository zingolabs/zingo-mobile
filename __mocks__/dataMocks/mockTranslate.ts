import { CurrencyEnum, DownloadMemosEnum, LanguageEnum } from '../../app/AppState';

export const mockTranslate = (p: string) => {
  if (p === 'about.copyright') {
    return String([
      '1 text translated line 1',
      '2 text translated line 2',
      '3 text translated line 3',
      '4 text translated line 4',
      '5 text translated line 5',
    ]);
  } else if (p === 'seed.buttontexts') {
    return `{
        "new": ["new"],
        "change": ["change"],
        "server": ["server"],
        "view": ["view"],
        "restore": ["restore"],
        "backup": ["backup"]
      }`;
  } else if (p === 'settings.memos') {
    return [
      {
        value: DownloadMemosEnum.noneMemos,
        text: 'text none',
      },
      {
        value: DownloadMemosEnum.walletMemos,
        text: 'text wallet',
      },
      {
        value: DownloadMemosEnum.allMemos,
        text: 'text all',
      },
    ];
  }
  if (p === 'settings.currencies') {
    return [
      {
        value: '',
        text: 'text no currency',
      },
      {
        value: CurrencyEnum.USDCurrency,
        text: 'text USD',
      },
    ];
  }
  if (p === 'settings.languages') {
    return [
      {
        value: LanguageEnum.en,
        text: 'text en',
      },
      {
        value: LanguageEnum.es,
        text: 'text es',
      },
      {
        value: LanguageEnum.pt,
        text: 'text pt',
      },
      {
        value: LanguageEnum.ru,
        text: 'text ru',
      },
    ];
  }
  if (p === 'settings.sendalls') {
    return [
      {
        value: true,
        text: 'text true',
      },
      {
        value: false,
        text: 'text false',
      },
    ];
  }
  if (p === 'settings.donations') {
    return [
      {
        value: true,
        text: 'text true',
      },
      {
        value: false,
        text: 'text false',
      },
    ];
  }
  if (p === 'settings.rescanmenus') {
    return [
      {
        value: true,
        text: 'text true',
      },
      {
        value: false,
        text: 'text false',
      },
    ];
  } else if (p === 'settings.recoverywalletinfoondevices') {
    return [
      {
        value: true,
        text: 'text true',
      },
      {
        value: false,
        text: 'text false',
      },
    ];
  } else {
    return 'text translated';
  }
};
