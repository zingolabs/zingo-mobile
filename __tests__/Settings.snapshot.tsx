/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import Settings from '../components/Settings';
import { defaultAppContextLoaded, ContextAppLoadedProvider } from '../app/context';
import { LanguageEnum, CurrencyEnum } from '../app/AppState';
import { mockTranslate } from '../__mocks__/dataMocks/mockTranslate';
import { mockInfo } from '../__mocks__/dataMocks/mockInfo';
import { mockTotalBalance } from '../__mocks__/dataMocks/mockTotalBalance';
import { mockServer } from '../__mocks__/dataMocks/mockServer';
import { mockWalletSettings } from '../__mocks__/dataMocks/mockWalletSettings';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.',
      groupingSeparator: ',',
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-community/netinfo', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RNCNetInfo = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
jest.mock('react-native-picker-select', () => 'RNPickerSelect');
jest.mock('react-native-keychain', () => ({
  ACCESS_CONTROL: {
    BIOMETRY_CURRENT_SET: 'biometryCurrentSet',
  },
  ACCESSIBLE: {
    WHEN_UNLOCKED_THIS_DEVICE_ONLY: 'whenUnlockedThisDeviceOnly',
  },
  AUTHENTICATION_TYPE: {
    BIOMETRICS: 'biometrics',
  },
  SECURITY_LEVEL: {
    SECURE_SOFTWARE: 'secureSoftware',
  },
  SECURITY_RULES: {
    NONE: 'none',
  },
  setGenericPassword: jest.fn(),
  getGenericPassword: jest.fn(),
  resetGenericPassword: jest.fn(),
}));
jest.useFakeTimers();

// test suite
describe('Component Settings - test', () => {
  //snapshot test
  const state = defaultAppContextLoaded;
  state.translate = mockTranslate;
  state.info = mockInfo;
  state.totalBalance = mockTotalBalance;
  state.server = mockServer;
  state.currency = CurrencyEnum.USDCurrency;
  state.language = LanguageEnum.en;
  state.sendAll = false;
  state.rescanMenu = false;
  state.recoveryWalletInfoOnDevice = true;
  state.donation = false;
  state.walletSettings = mockWalletSettings;
  const onClose = jest.fn();
  const onSetOption = jest.fn();
  test('Settings - snapshot', () => {
    const settings = render(
      <ContextAppLoadedProvider value={state}>
        <Settings
          closeModal={onClose}
          setWalletOption={onSetOption}
          setServerOption={onSetOption}
          setCurrencyOption={onSetOption}
          setLanguageOption={onSetOption}
          setSendAllOption={onSetOption}
          setDonationOption={onSetOption}
          setPrivacyOption={onSetOption}
          setModeOption={onSetOption}
          setSecurityOption={onSetOption}
          setSelectServerOption={onSetOption}
          setRescanMenuOption={onSetOption}
          set_recoveryWalletInfoOnDevice_option={onSetOption}
        />
      </ContextAppLoadedProvider>,
    );
    expect(settings.toJSON()).toMatchSnapshot();
  });
});
