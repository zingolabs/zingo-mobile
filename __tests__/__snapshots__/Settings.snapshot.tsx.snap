// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Settings - test Settings - snapshot 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "maxHeight": "85%",
      }
    }
    testID="settings.scroll-view"
  >
    <View>
      <View
        style={
          Object {
            "display": "flex",
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(28, 28, 30)",
              "fontSize": 18,
              "fontWeight": "bold",
              "opacity": 0.87,
            }
          }
        >
          text translated
        </Text>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "marginLeft": 25,
          }
        }
      />
      <View
        style={
          Object {
            "display": "flex",
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(28, 28, 30)",
              "fontSize": 18,
              "fontWeight": "bold",
              "opacity": 0.87,
            }
          }
        >
          text translated
        </Text>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "marginLeft": 25,
          }
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginRight": 10,
                "maxHeight": 50,
                "minHeight": 48,
                "opacity": 1,
              }
            }
            testID="settings.currency-"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <
                color="rgb(216, 216, 216)"
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [
                        128308,
                        128309,
                        128992,
                        128993,
                        128994,
                        128995,
                        128996,
                        9679,
                        9898,
                        9899,
                        11044,
                        61708,
                        61915,
                      ],
                      "f111",
                      "M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256z",
                    ],
                    "iconName": "circle",
                    "prefix": "far",
                  }
                }
                size={20}
              />
              <Text
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "marginLeft": 10,
                    "opacity": 1,
                  }
                }
              >
                text translated
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text no currency
          </Text>
        </View>
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginRight": 10,
                "maxHeight": 50,
                "minHeight": 48,
                "opacity": 1,
              }
            }
            testID="settings.currency-USD"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <
                color="rgb(216, 216, 216)"
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [
                        128280,
                        "dot-circle",
                      ],
                      "f192",
                      "M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-352a96 96 0 1 1 0 192 96 96 0 1 1 0-192z",
                    ],
                    "iconName": "circle-dot",
                    "prefix": "fas",
                  }
                }
                size={20}
              />
              <Text
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "marginLeft": 10,
                    "opacity": 1,
                  }
                }
              >
                text translated
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text USD
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(28, 28, 30)",
              "fontSize": 18,
              "fontWeight": "bold",
              "opacity": 0.87,
            }
          }
        >
          text translated
        </Text>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "marginLeft": 25,
          }
        }
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginRight": 10,
                "maxHeight": 50,
                "minHeight": 48,
                "opacity": 1,
              }
            }
            testID="settings.language-en"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <
                color="rgb(216, 216, 216)"
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [
                        128280,
                        "dot-circle",
                      ],
                      "f192",
                      "M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-352a96 96 0 1 1 0 192 96 96 0 1 1 0-192z",
                    ],
                    "iconName": "circle-dot",
                    "prefix": "fas",
                  }
                }
                size={20}
              />
              <Text
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "marginLeft": 10,
                    "opacity": 1,
                  }
                }
              >
                text translated
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text en
          </Text>
        </View>
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginRight": 10,
                "maxHeight": 50,
                "minHeight": 48,
                "opacity": 1,
              }
            }
            testID="settings.language-es"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "marginTop": 10,
                }
              }
            >
              <
                color="rgb(216, 216, 216)"
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [
                        128308,
                        128309,
                        128992,
                        128993,
                        128994,
                        128995,
                        128996,
                        9679,
                        9898,
                        9899,
                        11044,
                        61708,
                        61915,
                      ],
                      "f111",
                      "M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256z",
                    ],
                    "iconName": "circle",
                    "prefix": "far",
                  }
                }
                size={20}
              />
              <Text
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "marginLeft": 10,
                    "opacity": 1,
                  }
                }
              >
                text translated
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "opacity": 0.65,
              }
            }
          >
            text es
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginVertical": 5,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(0, 122, 255)",
          "borderColor": "rgb(28, 28, 30)",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "maxWidth": "90%",
          "minHeight": 48,
          "minWidth": "30%",
          "opacity": 1,
          "padding": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
      testID="settings.button.save"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 0,
            "minHeight": 48,
            "minWidth": 48,
            "padding": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(242, 242, 242)",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          text translated
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderColor": "rgb(0, 122, 255)",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "marginLeft": 10,
          "maxWidth": "90%",
          "minHeight": 48,
          "minWidth": "30%",
          "opacity": 1,
          "padding": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 0,
            "minHeight": 48,
            "minWidth": 48,
            "padding": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(0, 122, 255)",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          text translated
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
